lib/kcerts/ssh.go

```diff
@@ -173,26 +227,68 @@ func (a SSHAgent) GetEnv() []string {
 	return env
 }
 
-// FindSSHAgent Will start the ssh agent in the interactive terminal if it isn't present already as an environment variable
-// It will pull, in order: from the env, from the cache, create new.
+// FindSSHAgent will attempt to find a working ssh-agent socket, or will create
+// a new ssh-agent if necessary.  If a valid socket is found that does not
+// correspond with the desired standard location, a symlink is created.
+//
+// In detail:
+//   1) tries the socket indicated by the SSH_AUTH_SOCK environment variable.
+//   2) otherwise, tries the socket indicated by the cache.
+//   3) otherwise, starts a new ssh-agent.
+//   4) ensures the valid socket has the right path, otherwise does a symlink.
+//
+// The final ssh-agent socket returned by FindSSHAgent is always
+// ~/.config/enkit/agent.
 func FindSSHAgent(store cache.Store, logger logger.Logger) (*SSHAgent, error) {
+	var err error
 	agent := FindSSHAgentFromEnv()
-	if agent != nil && agent.Valid() {
-		return agent, nil
+	if agent != nil {
+		if !agent.Valid() {
+			logger.Warnf("%s from env isn't a valid ssh-agent socket.", agent.Socket)
+			agent = nil
+		}
 	}
-	agent, err := FetchSSHAgentFromCache(store)
-	if err != nil {
-		logger.Warnf("%s", err)
+
+	if agent == nil {
```

Comment:
the inverted `if agent == nil` logic is killing me :/ How about we break down `FindSSHAgent` in 2 functions:
1) `FindOrCreateAgent()` -> that just finds the agent, either from env, cache, or creates a new one? And is pretty much the old unmodified `FindSSHAgent` except it does NOT write to cache.
2) `FindSSHAgent` now invokes `FindOrCreateAgent`, and then does `UseStandardPaths` and `WriteAgentToCache`?

Bonus points if we rename this to `PrepareSSHAgent` or something that's more meaningful than `FindSSHAgent` - which is a misnomer at this point.

Reply:
Done